### PR TITLE
[Tests] Jormungandr starter refactor

### PR DESCRIPTION
**Pull request by dkijania**
Implemented remarks from https://github.com/input-output-hk/jormungandr/pull/983 with proposals to change bunch of methods into api.

### CODE REVIEW

File: jormungandr-integration-tests/src/common/jormungandr/starter.rs [hunk 1]
```diff
@@ -24,235 +24,224 @@ custom_error! {pub StartupError
...
+        process_assert::assert_process_failed_and_matches_message(command, &expected_msg);
+    }
+
+    fn success(&self) -> bool {
+        let rest_verifier = RestStartupVerification(self.config.clone());
+        let log_verifier = LogStartupVerification(self.config.clone());
+
+        match self.verification_mode {
+            StartupVerificationMode::Rest => rest_verifier.success(),
+            StartupVerificationMode::Log => log_verifier.success(),
```
Comment (mzabaluev) [inline jormungandr-integration-tests/src/common/jormungandr/starter.rs]: Small issue: both verifiers are initialized, but only one gets used.

File: jormungandr-integration-tests/src/common/jormungandr/starter.rs [hunk 2]
```diff
@@ -24,235 +24,224 @@ custom_error! {pub StartupError
...
-    let process = start_jormungandr_node_sync_with_retry(
-        &mut command,
-        config,
-        DEFAULT_SLEEP_BETWEEN_ATTEMPTS,
-        DEFAULT_MAX_ATTEMPTS,
-    );
-    process
-}
+#[derive(Clone, Debug)]
+pub struct RestStartupVerification(JormungandrConfig);
```
Comment (mzabaluev) [inline jormungandr-integration-tests/src/common/jormungandr/starter.rs]: Unless the struct is an obvious wrapper for the inner member, it's better to have the member named: `self.config` looks much clearer in the method code than `self.0`.

File: jormungandr-integration-tests/src/common/jormungandr/commands.rs [hunk 3]
```diff
@@ -45,23 +28,6 @@ pub fn get_start_jormungandr_as_leader_node_command(
...
-        .arg("--config")
-        .arg(config_path.as_os_str())
-        .arg("--genesis-block-hash")
-        .arg(&genesis_block_hash)
-        .stderr(get_stdio_from_log_file(&log_file_path));
-    println!("Running start jormungandr command: {:?}", &command);
-    command
-}
-
 pub fn get_start_jormungandr_as_passive_node_command(
```
[the remark this answers is not in the document]
Comment (mzabaluev) [reply jormungandr-integration-tests/src/common/jormungandr/commands.rs]: Not necessarily in this PR, though.

[the remark this answers is not in the document]
Comment (dkijania) [reply jormungandr-integration-tests/src/common/jormungandr/commands.rs]: Will implement it in some other pr

**Pull request merged**